runtime/compiler/compile/J9SymbolReferenceTable.cpp

```diff
@@ -880,11 +880,9 @@ J9::SymbolReferenceTable::findOrFabricateFlattenedArrayElementFieldShadowSymbol(
    const char *fieldName,
    const char *fieldSignature)
    {
-   int32_t flattenedFieldOffset = (int32_t)fieldOffset - (int32_t)TR::Compiler->om.objectHeaderSizeInBytes();
+   TR_ASSERT_FATAL(fieldOffset >= 0, "fieldOffset %d is invalid: fieldOffset %u objectHeaderSizeInBytes %" OMR_PRIuPTR " \n", fieldOffset, fieldOffset, TR::Compiler->om.objectHeaderSizeInBytes());
```

Comment:
As `fieldOffset` is unsigned, I believe the comparison `fieldOffset >= 0` will always be true.

Reply:
Good catch! I'll change the API to `int32_t fieldOffset`. 

I found [findOrFabricateFlattenedArrayElementFieldShadowSymbol is also defined in OMR](https://github.com/eclipse/omr/blob/0453d02bff7ff106a1ce429ffc7138c8711d19db/compiler/compile/OMRSymbolReferenceTable.hpp#L699) as well. I think it should be removed from OMR since it is not used.  I'll open a PR on it.

Reply:
Opened https://github.com/eclipse/omr/pull/6813

Reply:
Addressed in [3a216bd](https://github.com/eclipse-openj9/openj9/commit/3a216bd9dd80e89e99bdcdc61289366e65b5ed43)